modules/simple_forms_api/app/services/simple_forms_api/intent_to_file.rb

```diff
@@ -36,15 +42,23 @@ def submit
     end
 
     def existing_intents
-      @existing_intents ||= {
-        'compensation' => existing_compensation_intent,
-        'pension' => existing_pension_intent,
-        'survivor' => existing_survivor_intent
-      }
+      @existing_intents ||= if icn && participant_id
+                              {
+                                'compensation' => existing_compensation_intent,
+                                'pension' => existing_pension_intent,
+                                'survivor' => existing_survivor_intent
+                              }
+                            else
+                              {}
+                            end
     end
```

Comment:
This can be simplified slightly like so:

```ruby
def existing_intents
  return {} unless icn && participant_id

  @existing_intents ||= {
    'compensation' => existing_compensation_intent,
    'pension' => existing_pension_intent,
    'survivor' => existing_survivor_intent
  }
end
```